src/simulators/statevector/qubitvector.hpp

```diff
@@ -1188,6 +1152,47 @@ void QubitVector<data_t>::apply_matrix(const reg_t &qubits,
   } // end switch
 }
 
+template <typename data_t>
+void QubitVector<data_t>::apply_matrixes(const std::vector<reg_t> &qubitss,
+                                         const std::vector<cvector_t> &mats) {
+  if (mats.size() == 0)
+    return;
+
+  reg_t sorted_qubits;
+  for (const reg_t& qubits: qubitss)
```

Comment:
I would add a check here that if any set of qubits is > 2 it raises an error, rather than waiting to for the exception to be raised by the `expand_matrix` function.